### PR TITLE
Authentication

### PR DESCRIPTION
**Pull request by ZAINABFATIMA0**
Added user authentication through simple-jwt

### CODE REVIEW

File: users/serializer.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+from rest_framework import  serializers
+
+from .models import User
+
+
+class RegisterSerializer(serializers.ModelSerializer):
```
Comment (Hamza-Chaudhary65) [inline users/serializer.py]: shouldn't Serializer name reflect name of model it is serializing?

**Review by Hamza-Chaudhary65 — APPROVED** [review]

File: users/serializer.py (hunk 1 above)
Comment (ZAINABFATIMA0) [reply users/serializer.py]: I have updated the name to UserSerializer.

File: users/models.py [hunk 2]
```diff
@@ -7,7 +7,7 @@ class User(AbstractUser):
     email = models.EmailField(max_length=255, unique=True)
 
     USERNAME_FIELD = "email"
-    REQUIRED_FIELDS = []
+    REQUIRED_FIELDS = ["username"]
```
Comment (sheralim012) [inline users/models.py]: do we need this?

File: users/serializer.py [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+        model = User
+        fields = ('email', 'password', 'username', 'phone')
+        
+    def create(self, validated_data):
+        user = User(
+            email=validated_data['email'],
+            username=validated_data['username'],
+            phone=validated_data['phone']
+        )
+        user.set_password(validated_data['password'])
```
Comment (sheralim012) [inline users/serializer.py:29]: for registration, it's always better to have password and confirm password

File: users/urls.py [hunk 4]
```diff
@@ -1,3 +1,10 @@
 from django.urls import path
+from rest_framework_simplejwt import views as jwt_views
 
-urlpatterns = []
+from .views import register
+
+urlpatterns = [
+    path('register/', register, name='register'),
+    path('login/', jwt_views.TokenObtainPairView.as_view(), name='token_obtain_pair'),
+    path('login/refresh/', jwt_views.TokenRefreshView.as_view(), name='token_refresh')
```
Comment (sheralim012) [inline users/urls.py]:
```suggestion
    path('token-refresh/', jwt_views.TokenRefreshView.as_view(), name='token_refresh')
```

**Review by sheralim012 — CHANGES_REQUESTED** [review]

File: users/models.py (hunk 2 above)
Comment (ZAINABFATIMA0) [reply users/models.py]: I was getting this error previously "TypeError: create_superuser() missing 1 required positional argument: 'username'".
I can have either created the create_superuser method in UserManager that it doesn't require a username field. Alternatively, I added username to the required_fields in my user model. This lets me enter value for username when creating superuser.

**Review by Hamza-Chaudhary65 — APPROVED** [review]

File: users/models.py (hunk 2 above)
Comment (sheralim012) [reply users/models.py]: you need to create model manager

**Review by Hamza-Chaudhary65 — APPROVED** [review]

**Review by sheralim012 — APPROVED** [review]

**Pull request merged**